common/rpc/interceptor/mask_internal_error.go

```
@@ -42,51 +47,90 @@ var errorFrontendMasked = "something went wrong, please retry"
 	if ns.IsEmpty() {
 		return false
 	}
-	return i.maskInternalError(ns.String())
+	return mi.maskInternalError(ns.String())
 }
 
-func maskUnknownOrInternalErrors(err error) error {
-	st := serviceerror.ToStatus(err)
+func (mi *MaskInternalErrorDetailsInterceptor) maskUnknownOrInternalErrors(
+	req interface{}, fullMethodName string, err error,
+) error {
+	statusCode := serviceerror.ToStatus(err).Code()
 
-	if st.Code() != codes.Unknown && st.Code() != codes.Internal {
+	if statusCode != codes.Unknown && statusCode != codes.Internal {
 		return err
 	}
 
+	// we need to log the original error with hash.
+	// This code is similar to the one in telemetry.go
+
 	// convert internal and unknown errors into neutral error with hash code of the original error
 	errorHash := common.ErrorHash(err)
+	// logging the error with hash code
+	mi.logError(req, fullMethodName, err, errorHash, statusCode)
+
+	// returning masked error
 	maskedErrorMessage := fmt.Sprintf("%s (%s)", errorFrontendMasked, errorHash)
-	return status.New(st.Code(), maskedErrorMessage).Err()
+	return status.New(statusCode, maskedErrorMessage).Err()
+}
+
+func (mi *MaskInternalErrorDetailsInterceptor) logError(
+	req any,
+	fullMethod string,
+	err error,
+	errorHash string,
+	statusCode codes.Code,
+) {
+	methodName := api.MethodName(fullMethod)
+	overridedMethodName := TelemetryOverrideOperationTag(fullMethod, methodName)
+	nsName := MustGetNamespaceName(mi.namespaceRegistry, req)
+	var logTags []tag.Tag
+	if nsName == "" {
+		logTags = []tag.Tag{tag.Operation(overridedMethodName)}
+	} else {
+		logTags = []tag.Tag{tag.Operation(overridedMethodName), tag.WorkflowNamespace(nsName.String())}
+	}
+
+	logTags = append(logTags, tag.NewStringTag("hash", errorHash))
+
+	logTags = append(logTags, tag.NewStringTag("grpc_code", statusCode.String()))
+	logTags = append(logTags, mi.workflowTags.Extract(req, fullMethod)...)
+
+	mi.logger.Error("service failures", append(logTags, tag.Error(err))...)
```

Can we use a different error message here? so it would be easy to know if the log is from telemetry interceptor or mask error interceptor.

---

there will be a hash tag to differentiate. 
But yeah, just a filter will be nice. I will change to "masked service failures"